RNASeqPipeline.Rmd

```diff
@@ -6,139 +6,129 @@ output: pdf_document
+```{r simulate_data_fct}
+generateSequentialCountMatrix <- function(numHyp, varNum, diffExp, nSamples = 5){
   
-  runningMatrix <- matrix()
-  condList <- c()
-  diffExpression <- c()
+  # numHyp: ?
+  # varNum: ?
+  # diffExp: ?
   
-  for (i in 1:numHyp){
+  # initialize a list of datasets over time
+  synthData <- list()
   
-  synthData <- generateSyntheticData(dataset = "B_625_625", n.vars = varNum, 
-                                   samples.per.cond = 5, n.diffexp = diffExp, 
-                                   repl.id = 1, seqdepth = 1e7, 
-                                   fraction.upregulated = 0.5, 
-                                   between.group.diffdisp = FALSE, 
-                                   filter.threshold.total = -1, 
-                                   filter.threshold.mediancpm = -1, 
-                                   fraction.non.overdispersed = 0, 
-                                   output.file = "OutputTestSynth.rds")
+  #describe your conditions
+  condVec <- c(rep("control", nSamples), rep("treatment", nSamples)) 
   
- diffExpression <- c(diffExpression, synthData@variable.annotations$differential.expression)
-  
-  condList <- c(condList, "C","C", "C", "C","C","T","T","T","T","T") 
-  if (i == 1){
-    runningMatrix <- synthData@count.matrix
-  }
-  else{
-     runningMatrix <- cbind(runningMatrix, synthData@count.matrix)
-     
-     
-      
+  for (i in seq_along(1:numHyp)){
+    
+    # you need a seed for reproducibility of simulated data
+    set.seed(i)
+    
+    synthData[[i]] <- generateSyntheticData(dataset = "B_625_625", 
+                                            n.vars = varNum, 
+                                            samples.per.cond = nSamples, 
+                                            n.diffexp = diffExp, 
+                                            repl.id = 1, 
+                                            seqdepth = 1e7, 
+                                            fraction.upregulated = 0.5, 
+                                            between.group.diffdisp = FALSE, 
+                                            filter.threshold.total = -1, # don't understand this
+                                            filter.threshold.mediancpm = -1, # don't understand this
+                                            fraction.non.overdispersed = 0, 
+                                            output.file = "OutputTestSynth.rds")
+    
+    colnames(synthData[[i]]@count.matrix) <- condVec
   }
- 
-  
-  
   
+  # unless you plan on providing a different diffExp for each simulated dataset,
+  # we only need to save once
+  diffExpressionVec <- synthData[[1]]@variable.annotations$differential.expression
+
+  list(countMatrix = synthData,
+       sampleCondition = condVec,
+       diffExpression = diffExpressionVec)
 }
+```
+
+```{r}
+
+DESeqPipeline <- function(countMatrix, conditionList) {
   
-  for (i in 1 : dim(runningMatrix)[2]){
```

Comment:
don't need a separate for loop, can name within one for loop (see my revision), the more for loops you have, the slower your code